### ISSUE TITLE
[PY3] "Failed to read from defunct connection" when sending bytes

### ISSUE THREAD

**Issue by tardyp** [neo4j/neo4j-python-driver#147]
```python
driver = GraphDatabase.driver(url, auth=basic_auth(*auth))
with driver.session() as session:
    with session.begin_transaction() as tx:
        tx.run("CREATE (a:Person $map)", map=dict(bytes=b"xx"))
```
crashes with :
```
Traceback (most recent call last):
  File "test.py", line 10, in <module>
    tx.run("CREATE (a:Person $map)", map=dict(bytes=b"xx"))
  File ".venv/lib/python3.6/site-packages/neo4j/v1/api.py", line 492, in __exit__
    self.close()
  File ".venv/lib/python3.6/site-packages/neo4j/v1/api.py", line 562, in close
    self.sync()
  File ".venv/lib/python3.6/site-packages/neo4j/v1/api.py", line 534, in sync
    self.session.sync()
  File ".venv/lib/python3.6/site-packages/neo4j/v1/api.py", line 331, in sync
    detail_count, _ = self._connection.sync()
  File ".venv/lib/python3.6/site-packages/neo4j/bolt/connection.py", line 325, in sync
    detail_delta, summary_delta = self.fetch()
  File ".venv/lib/python3.6/site-packages/neo4j/bolt/connection.py", line 255, in fetch
    self._receive()
  File ".venv/lib/python3.6/site-packages/neo4j/bolt/connection.py", line 290, in _receive
    raise self.Error("Failed to read from defunct connection {!r}".format(self.server.address))
neo4j.exceptions.ServiceUnavailable: Failed to read from defunct connection Address(host='10.102.165.124', port=29627)
```
This confused me a lot, before I found out it is because of the bytes.
It's okay to only accept string, but I could have used some better erroring

**Comment by phidah** [comment]
Same problem arises if you try to send unicode.

**Comment by james-powis** [comment]
~~I am also seeing the same error when trying to set a property to a 5335 character long string... Not sure if it is related to this issue.~~

Disregard, I was sending the wrong value which was unicode

**Comment by tardyp** [comment]
@technige so was it fixed?

**Comment by technige** [comment]
@tardyp Yes, sorry, a note would have been helpful!

Support for bytes will become available in the v1.4 driver (https://github.com/neo4j/neo4j-python-driver/pull/169). Note that this will require a v3.2 or above server.
